### PR TITLE
chore: Repository Setup

### PR DESCRIPTION
**Pull request by cbolles**
* Add packages server, client, middleware
* Make NestJS starter project
* Update README
* Add GitHub Actions
* Add GitHub Issues Templates
* Add GitHub PR template

### CODE REVIEW

File: .github/workflows/client.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      run:
+        working-directory: packages/client
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
Comment (gregfrasco) [inline .github/workflows/client.yml]: Node 18?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Cargo Middleware: Typescript library which provides an S3 compliant middleware. The middleware communicates with the Cargo Server to generate the correct [AWS signature](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) if the user has the correct permissions. The user is identified via [JWT](https://jwt.io/).
+
+### Sample Workflows
+
+#### Getting User Permissions for a Specific Bucket
+
+Below is a sample workflow involving an application client and server. In this case the application is getting the user permissions for a specific bucket. The project ID is the unique ID for this specific application.
+
+```mermaid
```
Comment (gregfrasco) [inline README.md:27]: NOICE

**Review by gregfrasco — DISMISSED** [review]

File: .github/workflows/client.yml (hunk 1 above)
Comment (cbolles) [reply .github/workflows/client.yml]: Updated

File: README.md (hunk 2 above)
Comment (cbolles) [reply README.md:27]: It looks so good too!

**Review by gregfrasco — APPROVED** [review]

**Pull request merged**